src/test/java/com/adorsys/keycloakstatuslist/StatusListProtocolMapperTest.java

```diff
@@ -184,27 +185,26 @@ void shouldNotMap_IfCantPublishStatus() {
         assertThat(logCaptor.getErrorLogs(), hasItems(
                 containsString("Failed to publish status list %s".formatted(TEST_REALM_ID)),
                 containsString("Error publishing or updating status list on server"),
-                containsString("Failed to store index mapping")
-        ));
+                containsString("Failed to store index mapping")));
     }
 
     private void mockDefaultRealmConfig() {
         lenient().when(realm.getAttribute(StatusListConfig.STATUS_LIST_ENABLED)).thenReturn("true");
         lenient().when(realm.getAttribute(StatusListConfig.STATUS_LIST_SERVER_URL)).thenReturn(TEST_SERVER_URL);
     }
 
-    private long mockGetNextIndex() {
-        long nextIndex = new Random().nextLong();
-        Query query = mock(Query.class);
-        when(entityManager.createNativeQuery(startsWith("SELECT nextval"))).thenReturn(query);
-        when(query.getSingleResult()).thenAnswer(invocation -> nextIndex);
-        return nextIndex;
-    }
-
-    private void mockFailingGetNextIndex() {
-        Query query = mock(Query.class);
-        when(entityManager.createNativeQuery(startsWith("SELECT nextval"))).thenReturn(query);
-        when(query.getSingleResult()).thenThrow(new PersistenceException());
+    /**
+     * Mocks the EntityManager.persist() method.
+     * Use this instead of the old 'mockGetNextIndex'.
+     * This ensures the entity gets an ID and IDX assigned when the code saves it.
+     */
+    private void mockEntityPersist(long simulatedIndex) {
+        doAnswer(invocation -> {
+            StatusListMappingEntity entity = invocation.getArgument(0);
+            entity.setIdx(simulatedIndex); // Simulate sequence generation
+            entity.setId("test-uuid-" + simulatedIndex); // Simulate UUID generation
```

Comment:
Use module `UUID` to be more realistic.
```suggestion
            entity.setId(UUID.randomUUID().toString()); // Simulate UUID generation
```